engine/runtime/src/main/java/org/enso/interpreter/runtime/number/EnsoBigInteger.java

```diff
@@ -82,40 +83,50 @@ final boolean fitsInDouble() {
     return false;
   }
 
+  @ExportMessage
+  final boolean fitsInBigInteger() {
+    return true;
+  }
+
   @ExportMessage
   @CompilerDirectives.TruffleBoundary
   final byte asByte() throws UnsupportedMessageException {
-    return byteValue();
+    return value.byteValue();
   }
 
   @ExportMessage
   @CompilerDirectives.TruffleBoundary
   final short asShort() throws UnsupportedMessageException {
-    return shortValue();
+    return value.shortValue();
   }
 
   @ExportMessage
   @CompilerDirectives.TruffleBoundary
   final int asInt() throws UnsupportedMessageException {
-    return intValue();
+    return value.intValue();
   }
 
   @ExportMessage
   @CompilerDirectives.TruffleBoundary
   final long asLong() throws UnsupportedMessageException {
-    return longValue();
+    return value.longValue();
   }
 
   @ExportMessage
   @CompilerDirectives.TruffleBoundary
-  final float asFloat() throws UnsupportedMessageException {
-    return floatValue();
+  final float asFloat() {
+    return value.floatValue();
   }
 
   @ExportMessage
   @CompilerDirectives.TruffleBoundary
-  final double asDouble() throws UnsupportedMessageException {
-    return doubleValue();
+  public final double asDouble() {
+    return value.doubleValue();
```

Comment:
Out of curiosity, why do we need a truffle boundary here?

Reply:
`BigInteger.doubleValue()` is quite complex.